core/Symbols.cc

```diff
@@ -1207,17 +1207,23 @@ void Symbol::addLoc(const core::GlobalState &gs, core::Loc loc) {
     if (!loc.file().exists()) {
         return;
     }
-    for (auto &existing : locs_) {
-        if (existing.file() == loc.file()) {
-            existing = loc;
-            return;
+
+    if (ref(gs) != Symbols::root()) {
```

Comment:
ref is slow(but not as slow as the thing before).
WDYT about special casing it in caller to _not_ call addLoc on ref?
And an ENFORCE here that it should never be root?

Reply:
There's only one applicable callsite for that AFAIK, so that seems very reasonable.